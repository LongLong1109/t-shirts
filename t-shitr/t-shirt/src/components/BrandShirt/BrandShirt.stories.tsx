import type { Meta, StoryObj } from '@storybook/react'

import BrandShirt from '.'

const meta = {
  title: 'Components/BrandShirt',
  component: BrandShirt,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof BrandShirt>

export default meta
type Story = StoryObj<typeof BrandShirt>

export const BrandShirtPrimary: Story = {}
