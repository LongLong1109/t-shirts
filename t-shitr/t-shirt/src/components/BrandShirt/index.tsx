import Image from 'next/image'

// constants
import { brandList } from '@/constants/brandList'

const BrandShirt = () => (
  <section className='w-full flex-wrap items-center justify-center gap-10 bg-white-90 sm:flex sm:flex-col sm:gap-10 sm:items-center sm:p-10 md:flex-row'>
    {brandList.map((image, index) => (
      <div key={index}>
        <Image src={image.src} alt={image.alt} width={154} height={57} />
      </div>
    ))}
  </section>
)

export default BrandShirt
