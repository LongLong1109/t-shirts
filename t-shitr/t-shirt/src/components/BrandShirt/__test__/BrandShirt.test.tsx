import { render } from '@testing-library/react';

// component
import BrandShirt from '..';

describe('BrandShirt Component', () => {
  it('should matches snapshot', () => {
    const { asFragment } = render(<BrandShirt />);
    expect(asFragment()).toMatchSnapshot();
  });
});
