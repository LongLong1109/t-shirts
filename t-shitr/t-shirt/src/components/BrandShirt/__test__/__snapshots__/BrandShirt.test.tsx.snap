// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandShirt Component should matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="w-full flex-wrap items-center justify-center gap-10 bg-white-90 sm:flex sm:flex-col sm:gap-10 sm:items-center sm:p-10 md:flex-row"
  >
    <div>
      <img
        alt="basset"
        data-nimg="1"
        decoding="async"
        height="57"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="154"
      />
    </div>
    <div>
      <img
        alt="furniture"
        data-nimg="1"
        decoding="async"
        height="57"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="154"
      />
    </div>
    <div>
      <img
        alt="lazBoy"
        data-nimg="1"
        decoding="async"
        height="57"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="154"
      />
    </div>
    <div>
      <img
        alt="blackHawk"
        data-nimg="1"
        decoding="async"
        height="57"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="154"
      />
    </div>
    <div>
      <img
        alt="levizt"
        data-nimg="1"
        decoding="async"
        height="57"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
        style="color: transparent;"
        width="154"
      />
    </div>
  </section>
</DocumentFragment>
`;
